include/deal.II/matrix_free/constraint_info.h

```diff
@@ -176,6 +204,35 @@ namespace internal
 
       inline const typename Number::value_type *
       constraint_pool_end(const unsigned int row) const;
+
+      // TODO
+      std::vector<types::global_dof_index> local_dof_indices;
+      std::vector<types::global_dof_index> local_dof_indices_lex;
+      std::vector<ConstraintKinds>         mask;
+
+      template <typename T>
+      std::vector<std::vector<T>> &
+      get_dof_indices_per_cell()
+      {
+        if constexpr (std::is_same_v<T, unsigned int>)
+          return dof_indices_per_cell;
+        else if constexpr (std::is_same_v<T, types::global_dof_index>)
+          return dof_indices_per_cell_long;
+        else
+          return {};
```

Comment:
Is that last `else` what you want? Or do you really want to ensure that `T` can only either be `unsigned int` or `types::global_dof_index`?